snow/engine/common/tracker/peers.go

```diff
@@ -250,3 +271,14 @@ func (p *peerData) ConnectedPercent() float64 {
 func (p *peerData) SampleValidator() (ids.NodeID, bool) {
 	return p.connectedValidators.Peek()
 }
+
+func (p *peerData) GetValidators() set.Set[ids.NodeID] {
+	vdrs := set.Of(maps.Keys(p.validators)...)
+	return vdrs
+}
+
+func (p *peerData) ConnectedValidators() set.Set[ids.NodeID] {
+	// copy in case
+	vdrs := set.Of(p.connectedValidators.List()...)
+	return vdrs
+}
```

Comment:
`.List()` allocates a slice and then `set.Of` populates the map from the slice, `.Union` will just directly populate the map, without allocating the slice. So I think that would be a bit nicer.

Also I think the comment can be improved a bit.

```suggestion
func (p *peerData) ConnectedValidators() set.Set[ids.NodeID] {
	// The set is copied to avoid future changes from being reflected in the
	// returned set.
	copied := set.NewSet[ids.NodeID](len(p.connectedValidators))
	copied.Union(p.connectedValidators)
	return copied
}
```